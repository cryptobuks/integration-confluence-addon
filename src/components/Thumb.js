import React, { Component } from 'react'
import Analytics from '../helpers/Analytics'

class Thumb extends Component {
  constructor (props) {
    super(props)
    this.resizeImage = this.resizeImage.bind(this)
  }

  resizeImage () {
    window.AP.resize('100%', this.thumb.naturalHeight)
  }

  render () {
    const {previewImgUrl, projectName, shareUrl} = this.props
    return (
      <a
        href={shareUrl}
        target="_blank"
        rel="noopener noreferrer"
        onClick={ () => Analytics.track('Confluence.ThumbClicked') }>
        <img
          ref={el => { this.thumb = el }}
          src={previewImgUrl}
          alt={projectName}
          onLoad={this.resizeImage} />
      </a>
    )
  }
}

export default Thumb
