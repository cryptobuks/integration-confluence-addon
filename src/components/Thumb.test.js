import React from 'react'
import renderer from 'react-test-renderer';
import Thumb from './Thumb'

function createNodeMock(element) {
  if (element.type === 'img') {
    return {
      naturalHeight: '200'
    }
  }
  return null
}

it('renders correctly', () => {
  const options = {createNodeMock}
  const previewImgUrl = 'https://s3.invisionapp-cdn.com/storage.invisionapp.com/screens/thumbnails/235857145.png?x-amz-meta-iv=2&response-cache-control=max-age%3D2419200&x-amz-meta-mdi=0&x-amz-meta-ck=7b6ac614880f2a6dcb831703094006aa&AWSAccessKeyId=AKIAJFUMDU3L6GTLUDYA&Expires=1506816000&Signature=xSZfe1UhbXL%2FbsWq%2F5jn%2FuhT0Fc%3D'
  const projectName = 'Paper embed prototypes'
  const shareUrl = 'https://in.invisionapp.com/share/ATBWCTHR5'
  const component = renderer.create(
    <Thumb previewImgUrl={previewImgUrl} projectName={projectName} shareUrl={shareUrl} />,
    options
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  const img = tree.children[0]
  expect(img.type).toBe('img')
  global.AP = {
    resize: jest.fn()
  }
  img.props.onLoad()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  expect(global.AP.resize).toHaveBeenCalled()
})
