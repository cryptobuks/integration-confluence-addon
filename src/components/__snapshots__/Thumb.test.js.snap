// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="https://in.invisionapp.com/share/ATBWCTHR5"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Paper embed prototypes"
    onLoad={[Function]}
    src="https://s3.invisionapp-cdn.com/storage.invisionapp.com/screens/thumbnails/235857145.png?x-amz-meta-iv=2&response-cache-control=max-age%3D2419200&x-amz-meta-mdi=0&x-amz-meta-ck=7b6ac614880f2a6dcb831703094006aa&AWSAccessKeyId=AKIAJFUMDU3L6GTLUDYA&Expires=1506816000&Signature=xSZfe1UhbXL%2FbsWq%2F5jn%2FuhT0Fc%3D"
  />
</a>
`;

exports[`renders correctly 2`] = `
<a
  href="https://in.invisionapp.com/share/ATBWCTHR5"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Paper embed prototypes"
    onLoad={[Function]}
    src="https://s3.invisionapp-cdn.com/storage.invisionapp.com/screens/thumbnails/235857145.png?x-amz-meta-iv=2&response-cache-control=max-age%3D2419200&x-amz-meta-mdi=0&x-amz-meta-ck=7b6ac614880f2a6dcb831703094006aa&AWSAccessKeyId=AKIAJFUMDU3L6GTLUDYA&Expires=1506816000&Signature=xSZfe1UhbXL%2FbsWq%2F5jn%2FuhT0Fc%3D"
  />
</a>
`;
